### ISSUE TITLE
Feature compatibility tables for Hosted/Enterprise

### ISSUE THREAD

**Issue by schultyy** [travis-ci/docs-travis-ci-com#1652]
In the Enterprise support inbox we oftentimes get requests from customers where they run into problems because they try to use a feature which is only available on .org/.com, for instance:

- `sudo: required`
- `os: osx`
- `language: swift`
- `language: objective-c`

To minimize the confusion and also our support load I'd like to discuss if it's feasible to have compatibility tables for all major Travis CI features, indicating for which product/platform it is available.
MDN does something similar for JavaScript and CSS features, helping the user to find out if a certain feature is supported in their browser.
<img width="1024" alt="screen shot 2018-01-12 at 10 40 33" src="https://user-images.githubusercontent.com/235808/34868741-128be0b6-f785-11e7-9b1b-84a0e2869927.png">

I think that we could do this in a similar way. WDYT?

**Comment by acnagy** [comment]
Agreed - this would be helpful. Seems like a docs question to me - we have https://docs.travis-ci.com/user/reference/overview/ on Hosted, something similar on Enterprise would be good. Maybe on https://docs.travis-ci.com/user/enterprise? 

Shall we go ahead and open a docs repo issue to track this?

**Comment by danishkhan** [comment]
@schultyy are you thinking this table will be its own section in our documentation?

I wonder if it makes sense to reference Enterprise in specific sections of the existing documentation as well. For example, if I searched for how to enable Trusty build environments and found https://docs.travis-ci.com/user/reference/trusty/ I might try to use `sudo: required` since that's what the documentation says and maybe we should reference how things work with Enterprise on those pages as well in case our users stumble upon those pages from doing searches and don't go to a page that outlines the compatibilities or maybe we at least reference that page in all other pages where people might run into compatibility issues with Enterprise.

**Comment by schultyy** [comment]
@danishkhan That's a very good point!

I think your approach makes more sense than my initial thought having this feature compatibility table on all pages, because in the end it's probably just a handful of features which are either not available or work differently on Enterprise.
So that would lead us to `https://docs.travis-ci.com/user/enterprise` only having content on:
- Installation
- Operations manual
- Platform Administration tips 
- Upgrades
- Customizing build images
- ...
